subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/verification/DependencyVerificationSignatureWriteIntegTest.groovy

```diff
@@ -493,4 +497,72 @@ class DependencyVerificationSignatureWriteIntegTest extends AbstractSignatureVer
             assert keyRing.publicKey.userIDs.size() == 1
         }
     }
+
+    def "verification-keyring.keys is sorted and deduplicated by keyId"() {
+        def count = 50
+        def duplicatesCount = 10
+        def keyrings = Stream.generate { newKeyRing() }.limit(count - duplicatesCount).collect(Collectors.toList())
+        keyrings.addAll(keyrings.subList(0, duplicatesCount))
+
+        createMetadataFile {
+            keyServer(keyServerFixture.uri)
+        }
+
+        given:
+        javaLibrary()
+        def dependencies = new StringBuilder()
+        keyrings.eachWithIndex { keyring, index ->
+            uncheckedModule("org", "foo-$index", "1.0") {
+                withSignature {
+                    keyring.sign(it, [(SigningFixtures.validSecretKey): SigningFixtures.validPassword])
+                }
+            }
+            dependencies.append("implementation \"org:foo-$index:1.0\"\n")
+        }
+        buildFile << """
+            dependencies {
+                $dependencies
+            }
+        """
+
+        when:
+        //TODO: remove this once dependency verification stops triggering dependency resolution at execution time
```

Comment:
Maybe put an actual issue ID or link here, if there is one. Increases the chances of somebody finding this TODO when fixing that problem.

Reply:
Oh, Adam already removed it from the original, that solves the issues